httpclient/client.go

```diff
@@ -195,11 +195,13 @@ func (jc *HttpClient) doUploadFile(localPath, url string, httpClientsDetails htt
 
 	reqContent := fileutils.GetUploadRequestContent(file)
 	var reader io.Reader
+	var ok = false
 	if file != nil && progress != nil {
-		progressId := progress.New(size, "Uploading", localPath)
-		reader = progress.ReadWithProgress(progressId, reqContent)
-		defer progress.Abort(progressId)
-	} else {
+		progressReader := progress.NewReaderProgressBar(size, "Uploading", localPath)
+		reader, ok = progressReader.ActionWithProgress(reqContent).(io.Reader)
```

Comment:
the ok check can probably be removed if ActionWithProgress is changed (here and in similar places)